protocol/daemons/slinky/client/price_fetcher.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"cosmossdk.io/log"
+	"google.golang.org/grpc"
+
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+	"github.com/skip-mev/slinky/service/servers/oracle/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/pricefeed/api"
+	daemontypes "github.com/dydxprotocol/v4-chain/protocol/daemons/types"
+)
+
+// PriceFetcher is responsible for pulling prices from the slinky sidecar and sending them to the pricefeed server.
+type PriceFetcher interface {
+	Start(ctx context.Context) error
+	Stop()
+	FetchPrices(ctx context.Context) error
+}
+
+// PriceFetcherImpl implements the PriceFetcher interface.
+type PriceFetcherImpl struct {
+	marketPairFetcher      MarketPairFetcher
+	priceFeedServiceClient api.PriceFeedServiceClient
```

Comment:
what is the diff between priceFeedServiceClient, grpcClient, pricesConn?

Reply:
Can add docs here. The priceFeedServiceClient is the direct interface to the PriceFeedService which is where prices get posted. The grpcClient and pricesConn are the underlying connections which the priceFeedServiceClient is built on. They're needed solely to ensure connections are closed on shutdown.